### ISSUE TITLE
Broken reference counting when chaining refCount().first()

### ISSUE THREAD

**Issue by fterrazzoni** [ReactiveX/rxjs#1669]
**RxJS version:**
5.0.0-beta.6

**Code to reproduce:**
```js
subject = new Rx.Subject();
shared = subject.publishReplay(1).refCount();
shared.subscribe((x)=> console.log('Message:' + x));
subject.next('Hello');
shared.first().subscribe().unsubscribe();
subject.next('Goodbye');
```

**Expected behavior:**
```
Message:Hello
Message:Goodbye
```

**Actual behavior:**
```
Message:Hello
```

**Additional information:**

This code snippet behaves as expected in `RxJS 5 beta 3` and is broken since `RxJS 5 beta 4`. 

After some investigation, it looks like calling `shared.first().subscribe().unsubscribe()` on a `RefCountObservable` has the effect of decreasing `shared.refCount` by `1`, which is ultimately disconnecting the subject from the chain.

**Comment by blesh** [comment]
@trxcllnt this is directly related to the work you're currently doing on `ConnectableObservable`.

**Comment by trxcllnt** [comment]
resolved by #1705
